src/plugins/wizard/public/application/components/right_nav.tsx

```diff
@@ -32,19 +42,48 @@ export const RightNav = () => {
           options={options}
           valueOfSelected={activeVisName}
           onChange={(name) => {
-            dispatch(
-              setActiveVisualization({
-                name,
-                style: types.get(name)?.ui.containerConfig.style.defaults,
-              })
-            );
+            setNewVisType(name);
           }}
           fullWidth
+          data-test-subj="chartPicker"
         />
       </div>
       <div className="wizSidenav__style">
         <StyleSection />
       </div>
+      {newVisType && (
+        <EuiConfirmModal
+          title={i18n.translate('wizard.rightNav.changeVisType.modalTitle', {
+            defaultMessage: 'Change Visualization type',
+          })}
+          confirmButtonText={i18n.translate('wizard.rightNav.changeVisType.confirmText', {
+            defaultMessage: 'Ok',
+          })}
+          cancelButtonText={i18n.translate('wizard.rightNav.changeVisType.cancelText', {
+            defaultMessage: 'Cancel',
+          })}
+          onCancel={() => setNewVisType(undefined)}
+          onConfirm={() => {
+            dispatch(
+              setActiveVisualization({
+                name: newVisType,
+                style: types.get(newVisType)?.ui.containerConfig.style.defaults,
+              })
+            );
+
+            setNewVisType(undefined);
+          }}
+          maxWidth="300px"
+          data-test-subj="confirmVisChangeModal"
+        >
+          <p>
+            <FormattedMessage
+              id="wizard.rightNav.changeVisType.modalDescription"
+              defaultMessage="Currently, changing the visualization type clears the existing selection of fields. Are you sure you want to change the visualization type?"
```

Comment:
This is a bit wordy for my taste - I think we could condense quite a bit.
```suggestion
              defaultMessage="Changing the visualization type will reset all field selections. Are you sure you want to continue?"
```